Imagine/Filter/Loader/DownscaleFilterLoader.php

```diff
@@ -30,7 +30,12 @@ public function load(ImageInterface $image, array $options = array())
             $widthRatio = $width / $origWidth;
             $heightRatio = $height / $origHeight;
 
-            $ratio = min($widthRatio, $heightRatio);
+            // faster check than is_null
+            if ($width === null || $height === null) {
+                $ratio = max($widthRatio, $heightRatio);
+            } else {
+                $ratio = min($widthRatio, $heightRatio);
+            }
```

Comment:
could this be `$ratio = ($width === null || $height === null) ? max($widthRatio, $heightRatio) : min($widthRatio, $heightRatio);` ?